Src/PerformanceCollector/Filtering/CollectionConfigurationAccumulator.cs

```diff
@@ -0,0 +1,43 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Filtering
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    /// <summary>
+    /// Represents the operationalized metric-related component of the QuickPulse accumulator.
+    /// Unlike the main accumulator, this one might not have finished being processed at swap time,
+    /// so the consumer should keep the reference to it post-swap and make the best effort not to send
+    /// prematurely. <see cref="ReferenceCount"/> indicates that the accumulator is still being processed
+    /// when non-zero.
+    /// </summary>
+    internal class CollectionConfigurationAccumulator
+    {
+        /// <summary>
+        /// Used by writers to indicate that a processing operation is still in progress.
+        /// </summary>
+        public long ReferenceCount = 0;
```

Comment:
Consider protecting the ReferenceCount by exposing interlocked methods, not relying on callers to do it.